### PR TITLE
Documentation for textual service configuration

### LINKED ISSUE #4968: Document configuration dispatcher mechanism and feature
We are lacking documentation about the logic of the ConfigDispatcher, with automatic pid prefixing, exclusive definitions, etc.

We should add a short section in the docs, possible to the page https://www.eclipse.org/smarthome/documentation/features/dsl.html (while it should be renamed to something other than dsl) or next to that one.

@htreu Could you come up with something for that?

### PR DESCRIPTION
**Pull request by htreu**
Closes #4968.

Signed-off-by: Henning Treu <henning.treu@telekom.de>

### CODE REVIEW

File: docs/documentation/features/dsl.md [hunk 1]
```diff
@@ -134,3 +135,70 @@ Thing yahooweather:weather:losangeles [ location=2442047, unit="us", refresh=120
...
+All the above configuration files will be observed for changes.
+When the framework detects a change (or even a new *.cfg file) the changes will be read and applied to the specific service.
+Since a single service may be configured using multiple files this observation will _not delete_  configuration options from the Configuration Admin.
+In order to track deletion of configuration entries and files the configuration has to be marked as an exclusive service configuration.
+
+### Exclusive Service Configuration
+
+The framework will track exclusively marked service configurations for file or entry deletions. This way a configuration can be removed during runtime. For factory services even a whole service instance can be added and removed during runtime.
+
+To mark a configuration file exclusively for one service the _first line_ has to define the configuration PID in the format `pid:<configuration_pid>`.
```
Comment (lolodomo) [inline docs/documentation/features/dsl.md]: Is it possible to have an example ?

Comment (htreu) [reply docs/documentation/features/dsl.md]: yes, I will provide one.

File: docs/documentation/features/dsl.md [hunk 2]
```diff
@@ -7,6 +7,7 @@ layout: documentation
 # Textual Configuration
 
 Eclipse SmartHome provides the possibility to do fully text-based system setups. This is done using domain specific languages (DSLs) for the different kinds of artifacts.
+In addition, configuration files can be used to configure a lot of system OSGi services.
```
Comment (maggu2810) [inline docs/documentation/features/dsl.md]: Does it make sense to drop the words "a lot of system"? Can't we used it to configure all OSGi services that could be configured by configuration admin?

File: docs/documentation/features/dsl.md [hunk 3]
```diff
@@ -134,3 +135,86 @@ Thing yahooweather:weather:losangeles [ location=2442047, unit="us", refresh=120
...
+Since a single service may be configured using multiple files this observation will _not delete_  configuration options from the Configuration Admin.
+In order to track deletion of configuration entries and files the configuration has to be marked as an exclusive service configuration.
+
+### Exclusive Service Configuration
+
+The framework will track exclusively marked service configurations for file or entry deletions. This way a configuration can be removed during runtime. For factory services even a whole service instance can be added and removed during runtime.
+
+To mark a configuration file exclusively for one service the _first line_ has to define the configuration PID in the format `pid:<configuration_pid>`.
+By giving this PID marker the framework creates an exclusive configuration for the contents of this file.
+Other files without this marker which also define configuration for the given PID will be ignored for this PID.
```
Comment (triller-telekom) [inline docs/documentation/features/dsl.md]: also define configuration**s**

File: docs/documentation/features/dsl.md [hunk 4]
```diff
@@ -134,3 +135,86 @@ Thing yahooweather:weather:losangeles [ location=2442047, unit="us", refresh=120
...
+
+will exclusively configure the OSGi service with configuration_pid `org.eclipse.smarthome.bundle.myService` set.
+Any other configuration for this PID in other configuration files will be ignored now.
+In contrast, when removing the line `timeout=30` from the file, the service's `modified` method will be called with a new configuration which does not include the `timeout` option anymore.
+When removing the whole file the configuration will completely be deleted from the Configuration Admin.
+
+
+### Factory Service Configuration
+
+Using the format of an exclusive service configuration it is also possible to create several instances of a specific services.
```
Comment (triller-telekom) [inline docs/documentation/features/dsl.md]: of a specific service~~s~~

File: docs/documentation/features/dsl.md (hunk 2 above)
Comment (htreu) [reply docs/documentation/features/dsl.md]: thats, correct. Thanks.

**Pull request merged**